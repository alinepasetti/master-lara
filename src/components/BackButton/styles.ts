import styled, { css } from 'styled-components';
import { BiArrowBack } from 'react-icons/bi';

export const Back = styled(BiArrowBack)`${({ theme }) => css`
  width: 40px;
  height: 40px;
  color: ${theme.colors.mediumGray};
  background-color: ${theme.colors.white};
  font-size: 0.8rem;
  border-radius: 50%;
  padding: 0.7rem;
  position: absolute;
  left: 1.6rem;
  top: 1.6rem;
`}`;
