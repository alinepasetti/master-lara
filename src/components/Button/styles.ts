import styled, { css } from 'styled-components';

export const Button = styled.a` ${({ theme }) => css`
  width: 90%;
  display: block;
  padding: 1.6rem;
  font-size: 2.4rem;
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
  border: none;
  margin: 0 auto 2.4rem;
  text-align: center;
  text-decoration: none;
`}`;
