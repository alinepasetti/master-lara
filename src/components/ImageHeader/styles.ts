import styled from 'styled-components';

export const Image = styled.img`
  width: 100%;
  height: 40vh;
  object-fit: cover;
  object-position: center;
`;
