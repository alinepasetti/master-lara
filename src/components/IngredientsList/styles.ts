import styled, { css } from 'styled-components';
import { AiFillCloseCircle } from 'react-icons/ai';
import { AiFillCheckCircle } from 'react-icons/ai';

type HasIngredient = {
  hasIngredient: boolean;
};

const uncheckIconUrl = (theme) =>
  `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='%23${theme.colors.secondaryColor.substring(
    1,
  )}' class='w-6 h-6'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' d='M9.75 9.75l4.5 4.5m0-4.5l-4.5 4.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z' /%3E%3C/svg%3E")`;

const checkIconUrl = (theme) =>
  `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 24 24' stroke-width='1.5' stroke='%23${theme.colors.green.substring(
    1,
  )}' class='w-6 h-6'%3E%3Cpath stroke-linecap='round' stroke-linejoin='round' d='M9 12.75L11.25 15 15 9.75M21 12a9 9 0 11-18 0 9 9 0 0118 0z' /%3E%3C/svg%3E%0A")`;

export const Container = styled.ul`
  padding: 2.4rem;
  padding-top: 0;
  margin: 0;

  &> a {
    margin-top: 1.6rem;
  }
`;

export const ListItem = styled.li<HasIngredient>`${({
  theme,
  hasIngredient,
}) => css`
  list-style: none;

  &::before {
    display: inline-block;
    content: ${hasIngredient ? checkIconUrl(theme) : uncheckIconUrl(theme)};
    width: 1.8rem;
    padding-right: 1rem;
}
`}`;

export const CheckIcon = styled(AiFillCheckCircle)`${({ theme }) => css`
  color: ${theme.colors.green};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;

export const UncheckIcon = styled(AiFillCloseCircle)`${({ theme }) => css`
  color: ${theme.colors.secondaryColor};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;
