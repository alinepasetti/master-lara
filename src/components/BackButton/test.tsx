import { screen } from '@testing-library/react';
import { BackButton } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<BackButton />', () => {
  it('should render a link', () => {
    renderTheme(<BackButton />);
    const link = screen.getByTestId('back');
    expect(link).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    renderTheme(<BackButton />);
    expect(screen.getByTestId('back')).toMatchSnapshot();
  });
});
