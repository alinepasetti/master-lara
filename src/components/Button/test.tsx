import { screen } from '@testing-library/react';
import { Button } from '.';
import { renderTheme } from '../../styles/render-theme';
import { vi } from 'vitest';

describe('<Button />', () => {
  it('should render a button', () => {
    renderTheme(<Button clickHandler={() => 'hi'}>Example</Button>);
    const button = screen.getByText('Example');
    expect(button).toBeInTheDocument();
  });

  it('should call the click handler', () => {
    const clickHandler = vi.fn();
    renderTheme(<Button clickHandler={clickHandler}>Example</Button>);
    const button = screen.getByText('Example');
    button.click();
    expect(clickHandler).toBeCalledTimes(1);
  });

  it('should match snapshot', () => {
    renderTheme(<Button clickHandler={() => 'hi'}>Example</Button>);
    expect(screen.getByText('Example')).toMatchSnapshot();
  });
});
