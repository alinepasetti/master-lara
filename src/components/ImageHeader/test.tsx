import { screen } from '@testing-library/react';
import { ImageHeader } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<ImageHeader />', () => {
  it('should render a ImageHeader', () => {
    renderTheme(<ImageHeader text="image text" srcImage="image.jpg" />);
    expect(screen.getByAltText('image text')).toHaveAttribute(
      'src',
      'image.jpg',
    );
  });
});
