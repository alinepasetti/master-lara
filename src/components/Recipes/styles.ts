import { Title as Message } from 'components/Heading/styles';
import styled from 'styled-components';

export const Container = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding-bottom: 2.4rem;
  &> ${Message} {
    text-align: center;
    margin: 0 2.4rem;
  }
`;

export const Divider = styled.hr`
  border: 0;
  border-bottom: 1px solid #c6c9cd;
  margin: 2.4rem auto 0 auto;
  width: 89%;
`;
