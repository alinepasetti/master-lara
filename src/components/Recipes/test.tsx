import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { RecipesContext } from '../../contexts/RecipesProvider/context';
import { RequestStatus } from '../../contexts/RecipesProvider/types';
import { RECIPE_MOCK_DATA } from '../RecipeCard/test';
import { Recipes } from '.';

describe('<Recipes />', () => {
  it('should render the loading label', () => {
    renderTheme(
      <RecipesContext.Provider
        value={{
          recipes: [],
          searchedIngredients: [],
          requestStatus: RequestStatus.RECIPES_LOADING,
        }}
      >
        <Recipes />
      </RecipesContext.Provider>,
    );
    const heading = screen.getByRole('heading', { name: /loading/i });
    expect(heading).toBeInTheDocument();
  });

  it('should render 3 recipe cards', () => {
    renderTheme(
      <RecipesContext.Provider
        value={{
          recipes: [RECIPE_MOCK_DATA, RECIPE_MOCK_DATA, RECIPE_MOCK_DATA],
          searchedIngredients: [],
          requestStatus: RequestStatus.RECIPES_SUCCESS,
        }}
      >
        <Recipes />
      </RecipesContext.Provider>,
    );
    const recipeCards = screen.getAllByRole('link', { name: /sweet sugar/i });
    expect(recipeCards).toHaveLength(3);
  });

  it('should match snapshot', () => {
    renderTheme(
      <RecipesContext.Provider
        value={{
          recipes: [RECIPE_MOCK_DATA, RECIPE_MOCK_DATA, RECIPE_MOCK_DATA],
          searchedIngredients: [],
          requestStatus: RequestStatus.RECIPES_SUCCESS,
        }}
      >
        <Recipes />
      </RecipesContext.Provider>,
    );
    expect(
      screen.getAllByRole('link', { name: /sweet sugar/i })[0].parentElement,
    ).toMatchSnapshot();
  });
});
