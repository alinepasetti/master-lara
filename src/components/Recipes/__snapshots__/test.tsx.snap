// Vitest Snapshot v1

exports[`<RecipeCard /> > should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 90%;
  margin: 2em auto;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  max-width: 100%;
  width: 100%;
  height: 35vh;
  border-radius: 8px;
  object-fit: cover;
  object-position: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.8rem 0;
}

.c3 {
  font-size: 1.6rem;
  margin: 0 0;
  color: #5c5c5c;
  font-weight: 400;
}

.c4 {
  font-size: 1.6rem;
  margin: 0 0;
  color: #5c5c5c;
  font-weight: 400;
}

<a
  class="c0"
  href="/recipe/id"
>
  <img
    alt="Sweet Sugar"
    class="c1"
    src="image.jpg"
  />
  <div
    class="c2"
  >
    <h1
      class=""
    >
      Sweet Sugar
    </h1>
    <h3
      class="c3"
    >
      Indian
    </h3>
    <h3
      class="c3"
    >
      1 hour
    </h3>
    <h3
      class="c4"
    >
      1 / 3
      <span>
        ingredients
      </span>
    </h3>
  </div>
</a>
`;

exports[`<Recipes /> > should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 90%;
  margin: 2em auto;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  max-width: 100%;
  width: 100%;
  height: 35vh;
  border-radius: 8px;
  object-fit: cover;
  object-position: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.8rem 0;
}

.c3 {
  font-size: 1.6rem;
  margin: 0 0;
  color: #5c5c5c;
  font-weight: 400;
}

.c4 {
  font-size: 1.6rem;
  margin: 0 0;
  color: #5c5c5c;
  font-weight: 400;
}

<div
  class=""
>
  <a
    class="c0"
    href="/recipe/id"
  >
    <img
      alt="Sweet Sugar"
      class="c1"
      src="image.jpg"
    />
    <div
      class="c2"
    >
      <h1
        class=""
      >
        Sweet Sugar
      </h1>
      <h3
        class="c3"
      >
        Indian
      </h3>
      <h3
        class="c3"
      >
        1 hour
      </h3>
      <h3
        class="c4"
      >
        1 / 3
        <span>
          ingredients
        </span>
      </h3>
    </div>
  </a>
  <a
    class="c0"
    href="/recipe/id"
  >
    <img
      alt="Sweet Sugar"
      class="c1"
      src="image.jpg"
    />
    <div
      class="c2"
    >
      <h1
        class=""
      >
        Sweet Sugar
      </h1>
      <h3
        class="c3"
      >
        Indian
      </h3>
      <h3
        class="c3"
      >
        1 hour
      </h3>
      <h3
        class="c4"
      >
        1 / 3
        <span>
          ingredients
        </span>
      </h3>
    </div>
  </a>
  <a
    class="c0"
    href="/recipe/id"
  >
    <img
      alt="Sweet Sugar"
      class="c1"
      src="image.jpg"
    />
    <div
      class="c2"
    >
      <h1
        class=""
      >
        Sweet Sugar
      </h1>
      <h3
        class="c3"
      >
        Indian
      </h3>
      <h3
        class="c3"
      >
        1 hour
      </h3>
      <h3
        class="c4"
      >
        1 / 3
        <span>
          ingredients
        </span>
      </h3>
    </div>
  </a>
</div>
`;
